security/pkg/pki/ca/ca.go

```diff
@@ -183,11 +183,15 @@ func NewSelfSignedIstioCAOptions(ctx context.Context,
 		// Write the key/cert back to secret so they will be persistent when CA restarts.
 		secret := k8ssecret.BuildSecret("", CASecret, namespace, nil, nil, nil, pemCert, pemKey, istioCASecretType)
 		if _, err = client.Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{}); err != nil {
-			pkiCaLog.Errorf("Failed to write secret to CA (error: %s). Abort.", err)
-			return nil, fmt.Errorf("failed to create CA due to secret write error")
+			// Ignore error if secret got created by another replica
```

Comment:
won't we have the wrong data though? we are storing or data at L200 but its not the right key

Reply:
The PR may cause inconsistence between the CA key and CA bundle. The following is an example sequence of how the inconsistence occurs.
- Controller A and B both try to create a CA secret.
- A succeeds and B fails at creating a CA secret.
- B reads the secret created by A successfully and proceed to line 200 to write B's CA bundle.
- B writes CA bundle successfully, which is inconsistent with the CA secret created by A.

Reply:
@lei-tang : step number 4 in your sequence - line 187 is overwriting the caSecret with the data read from A in case of conflict, and that will be used at line 200, won't that help?

Reply:
Another inconsistency under such scenario is the rootCerts on line 192 is B's rootCerts whereas the CA secret is A's CA secret.
line 192  pkiCaLog.Infof("Using self-generated public key: %v", string(rootCerts))

Reply:
The line 192 uses rootCerts from B at line 174, whereas at line 196, rootCerts is read from CA secret generated by A.

Reply:
ack about the logging inconsistency